vllm/executor/ray_gpu_executor.py

```diff
@@ -224,16 +224,13 @@ def _init_workers_ray(self, placement_group: "PlacementGroup",
         # broadcasted to.
         self.non_driver_workers: List[RayWorkerWrapper] = []
 
-        for pp_rank in range(self.parallel_config.pipeline_parallel_size):
-            for tp_rank in range(self.parallel_config.tensor_parallel_size):
-                rank = (pp_rank *
-                        self.parallel_config.tensor_parallel_size) + tp_rank
-                if rank == 0:
-                    pass
-                elif rank % self.parallel_config.tensor_parallel_size == 0:
-                    self.tp_driver_workers.append(self.workers[rank - 1])
-                else:
-                    self.non_driver_workers.append(self.workers[rank - 1])
+        for idx, rank in enumerate(worker_ranks[1:]):
```

Comment:
we can iterate over `worker_ranks`? I think `worker_ranks[0] == 0` is always true.

Reply:
Yes, but then we need to do `idx - 1`. It's a little bit less clean. We can get rid of the first `if` condition as well if you like that style more, but I kept it for readability.

Reply:
let's get rid of the first if condition as well, and put a comment here, saying that `worker_ranks[0] == 0` .

Reply:
Done